compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -611,7 +611,14 @@ public void visit(BLangAnnotationAttachment annAttachmentNode) {
 
     public void visit(BLangSimpleVariable varNode) {
 
+        boolean configurable = isConfigurable(varNode);
+
         if (varNode.isDeclaredWithVar) {
+            // Configurable variable cannot be declared with var
+            if (configurable) {
+                dlog.error(varNode.pos, DiagnosticCode.CONFIGURABLE_VARIABLE_CANNOT_BE_DECLARED_WITH_VAR);
+                return;
```

Comment:
We shouldn't return here. Continue with the error. Otherwise, we will not type check RHS of the variable.

Reply:
thanx will change